### ISSUE TITLE
[BUG] Deprecation warning in docstrings

### ISSUE THREAD

**Issue by mbeyeler** [pulse2percept/pulse2percept#249]
The deprecation warning for functions is formatted wrong. Use a `deprecated` directive instead.

`pulse2percept.utils.deprecation.py`:

```
    def _update_doc(self, old_doc, msg=None):
        """Updates the docstring"""
        if msg is None:
            msg = self._get_message("This feature")
        # Insert a deprecated directive:
        doc = ".. deprecated:: %s\n\n    %s" % (self.deprecated_version, msg)
        if old_doc:
            doc = "%s\n\n%s" % (doc, old_doc)
        return doc

    def _decorate_class(self, cls):
        """Mark a class as deprecated"""
        msg = self._get_message("Class %s" % cls.__name__)

        # FIXME: we should probably reset __new__ for full generality
        init = cls.__init__

        def wrapped(*args, **kwargs):
            warnings.warn(msg, category=DeprecationWarning)
            return init(*args, **kwargs)
        cls.__init__ = wrapped

        wrapped.__name__ = '__init__'
        wrapped.__doc__ = self._update_doc(init.__doc__, msg)
        wrapped.deprecated_original = init

        return cls

    def _decorate_property(self, prop):
        """Mark a class property as deprecated

        Note that this is only triggered properly if the `property` decorator
        comes before the `deprecated` decorator, like so:

        .. code-block:: python

            @deprecated()
            @property
            def deprecated_attribute_(self):
                ...
        """
        msg = self._get_message("Property %s" % prop.__name__)

        @property
        def wrapped(*args, **kwargs):
            warnings.warn(msg, category=DeprecationWarning)
            return prop.fget(*args, **kwargs)

        wrapped.__doc__ = self._update_doc(prop.__doc__, msg)
        return wrapped

    def _decorate_fun(self, fun):
        """Mark a function as deprecated"""
        msg = self._get_message("Function %s" % fun.__name__)

        @functools.wraps(fun)
        def wrapped(*args, **kwargs):
            warnings.warn(msg, category=DeprecationWarning)
            return fun(*args, **kwargs)

        wrapped.__doc__ = self._update_doc(wrapped.__doc__, msg)

        return wrapped
```

**Comment by aiwenxu** [comment]
I am going to work on this

**Comment by mbeyeler** [comment]
Closed via #255
